modules/gdscript/gdscript_parser.cpp

```diff
@@ -7842,13 +7842,16 @@ void GDScriptParser::_check_block_types(BlockNode *p_block) {
 							return;
 						}
 
-						if (!lh_type.has_type && check_types) {
-							if (op->arguments[0]->type == Node::TYPE_OPERATOR) {
-								_mark_line_as_unsafe(op->line);
+						if (check_types) {
+							if (!lh_type.has_type) {
+								if (op->arguments[0]->type == Node::TYPE_OPERATOR) {
+									_mark_line_as_unsafe(op->line);
+								}
+							}
+							if (lh_type.is_constant) {
```

Comment:
What's surprising is that `lh_type` evaluates as a constant when it's a local variable though.

Reply:
I don't think I managed to reproduce that one, but, maybe it fixed the others.

Reply:
Not sure why it would be marked as constant. But the fix looks good to me.

Reply:
I guess that was due to b24cb922?